jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -441,4 +463,30 @@ public void triggerAdHocAction(String containerId,
                                                                                              PAGE_SIZE_UNLIMITED);
         return processDefinitions.stream().map(new ProcessDefinitionMapper()).collect(toList());
     }
+
+    public static class HumanTaskNodePredicate implements BiPredicate<CaseActionSummary, List<CaseActionSummary>> {
+
+        @Override
+        public boolean test(final CaseActionSummary summary,
+                            final List<CaseActionSummary> actions) {
+            return NODE_TYPE_HUMAN_TASK.contains(summary.getType()) && actions.stream().filter(action -> action.getType().equals(summary.getType()) && action.getName().equals(summary.getName())).findAny().isPresent();
+        }
+    }
+
+    public static class MilestoneNodePredicate implements Predicate<CaseActionSummary> {
+
+        @Override
+        public boolean test(final CaseActionSummary summary) {
+            return NODE_TYPE_MILESTONE.equals(summary.getType());
+        }
+    }
+
+    public static class MilestonePredicate implements BiPredicate<CaseActionSummary, String> {
```

Comment:
I'm failing to the see the value in introducing this ```BiPredicate```.
It's sole use is in ```findMilestone()``` method as far as I can see, which is called only when the expression 
```new MilestoneNodePredicate().negate().test(action)``` has already been evaluated to ```false```, which means the part 
```new MilestoneNodePredicate().test(summary)``` in BiPredicate's test method will always evaluate to ```true```.
In that case the whole filtering in ```findMilestone()``` can be simplified to comparing names only.

Reply:
Completly agree, removed ```MilestoneNodePredicate```